packages/ckeditor5-code-block/src/outdentcodeblockcommand.ts

```diff
@@ -179,6 +179,19 @@ function getCodeLineTextNodeAtPosition( position: Position ): Text | null {
 		nodeAtPosition = position.nodeBefore;
 	}
 
+	// <codeBlock>foo <inlineElement></inlineElement>^</codeBlock>
+	// <codeBlock><inlineElement></inlineElement>^</codeBlock>
+	if ( nodeAtPosition && nodeAtPosition.is( 'element' ) && !nodeAtPosition.is( 'element', 'softBreak' ) ) {
+		while ( nodeAtPosition.previousSibling !== null && !nodeAtPosition.previousSibling.is( 'element', 'softBreak' ) ) {
```

Comment:
Shouldn't this condition also check for text nodes? I mean, AFAIU, you want to jump left until you find:

*   start of code block - it means we were in the first line and there was no text before inline element(s),
*   soft break - it means that we are in 2+ line and there was no text before inline element(s),
*   text - it means there was some text before inline element(s) and we should take it.

Now it looks to me that you jump over text nodes until you find a soft break (if you find it).

So it seems to me this should work incorrectly for case like this:

`<codeBlock>    foo<inlineElement></inlineElement>^</codeBlock>`

If I am not mistaken, this will return `null` while it should return  `foo` text node, right?